crates/sui-json-rpc/src/read_api.rs

```diff
@@ -872,12 +872,13 @@ impl ReadApiServer for ReadApi {
                 .into_iter()
                 .enumerate()
                 .map(|(seq, e)| {
+                    let layout = store.executor().type_layout_resolver(Box::new(state.get_db())).get_annotated_layout(&e.type_)?;
```

Comment:
Ideally we would create the layout resolver once in `get_events` and then use it multiple times in the `map` here.  I realise that the lack of a `Send` constraint on the `dyn` type returned by `type_layout_resolver` may be an impediment for that, so we can follow up on that fix perhaps? (Otherwise you may be able to do the same trick as you used elsewhere which was to factor the non-async part into its own function).

Reply:
what's the reasoning that we need to have this be a dyn type vs a concrete but maybe generic one?

Reply:
Its implementation is returned by the execution layer, so its underlying implementation could change during reconfiguration.